src/Runners/PHPUnit/BaseRunner.php

```diff
@@ -95,11 +98,28 @@ private function load(SuiteLoader $loader): void
         $this->pending = $this->options->functional()
             ? $loader->getTestMethods()
             : $loader->getSuites();
+
+        $this->sortPending();
+
         foreach ($this->pending as $pending) {
             $this->printer->addTest($pending);
         }
     }
 
+    private function sortPending(): void
+    {
+        if ($this->options->orderBy() === Options::ORDER_RANDOM) {
+            mt_srand($this->options->randomOrderSeed());
+            shuffle($this->pending);
+        }
```

Comment:
randomness may also affect tests, so `mt_srand` should not be called the second time without the seed.